server/test/handlers/availability.test.js

```diff
@@ -4,40 +4,197 @@ describe('testing fetching of worker availability functionality', () => {
+
+        // Assert
+        expect(availableTimes).toBe(null);
+    });
+
+    //TEST 7
+    test('rejection of empty string value for required schoolId parameter', async () => {
+        // Arrange
+        const workerId = 8000000;
+        const schoolId = '';
+        const startTime = '8:00:00';
+        const endTime = '8:30:00';
+
+        // Act
+        const availableTimes = await availabilityHandler.getAvailabilityDetails(workerId, schoolId, startTime, endTime);
+
+        // Assert
+        expect(availableTimes).toBe(null);
+    });
+    //TEST 8
+
+    test('rejection of empty string value for required startTime parameter', async () => {
+        // Arrange
+        const workerId = 8000000;
+        const schoolId = 1;
+        const startTime = '';
+        const endTime = '8:30:00';
+
+        // Act
+        const availableTimes = await availabilityHandler.getAvailabilityDetails(workerId, schoolId, startTime, endTime);
+
+        // Assert
+        expect(availableTimes).toBe(null);
+    });
+    //TEST 9
+    test('rejection of empty string value for required endTime parameter', async () => {
         // Arrange
         const workerId = 8000000;
         const schoolId = 1;
-        const startTime = '2020-10-20';
-        const endTime = '2020-10-20';
-
-        const testAvailableTimes = [
-            {
-                workerTimeslotId: 1,
-                startTime: '08:00:00', 
-                endTime: '08:30:00',
-                date: '2020-10-20'
-            },
-            {
-                workerTimeslotId: 2,
-                startTime: '08:30:00', 
-                endTime: '09:00:00',
-                date: '2020-10-20'
-            },
-            {
-                workerTimeslotId: 3,
-                startTime: '09:00:00', 
-                endTime: '09:30:00',
-                date: '2020-10-20'
-            }
-        ];
-
-        // Act
-        const availableTimes = await availabilityHandler.getWorkerAvailability(workerId, schoolId, startTime, endTime);
-
-        // Assert
-        expect(availableTimes).toMatchObject(testAvailableTimes);
+        const startTime = '8:00:00';
+        const endTime = '';
+
+        // Act
+        const availableTimes = await availabilityHandler.getAvailabilityDetails(workerId, schoolId, startTime, endTime);
+
+        // Assert
+        expect(availableTimes).toBe(null);
     });
+    /*
+    //TEST 10
+    test('rejection of invalid workerId', async () => {
+     // Arrange
+     const workerId = 0;
+     const schoolId = 1;
+     const startTime = '8:00:00';
+     const endTime = '8:30:00';
+    
+     // Act
+     const availableTimes = await availabilityHandler.getAvailabilityDetails(workerId, schoolId, startTime, endTime);
+    
+     // Assert
+     expect(availableTimes).toBe(null);
+    });
+    */
+
+    /* The following code is the test for the stub which will be deleted once this PR is approved.
```

Comment:
Hi Nancy, 
I would recommend deleting these lines to avoid future confusion since they can always be copied from the past PR if needed.

Reply:
Hi Amy,
Thank you for suggesting that!
My action: I've removed the hard-coded test conditions on line 146 onwards of the server/test/handlers/availability.test.js

Please confirm if this takes care of your suggestion!